index.js

```diff
@@ -360,8 +360,13 @@ Command.prototype.action = function(fn) {
     fn.apply(self, actionArgs);
   };
   var parent = this.parent || this;
-  var name = parent === this ? '*' : this._name;
-  parent.on('command:' + name, listener);
+  if (parent === this) {
+    parent.on('program-command', listener);
+    parent.on('command:*', listener);
```

Comment:
Ah no, I think should just be adding listener for `'program-command'` and not `'command:*'` or the listener will get run twice.

The listeners for `command:*` get created when caller explicitly adds one (e.g. for detecting unknown commands) or caller adds a command called `*` (i.e. `.command('*')`

Reply:
```suggestion
```